### PR TITLE
Solr timestamps

### PR DESCRIPTION
**Pull request by dannymandel**
Add timestamps from original source collection and timestamp when we imported a record

### CODE REVIEW

**Comment by dannymandel** [comment]
Tested it out locally for SESAR and OpenContext, as those are the ones that have the data available, confirmed I was able to find it:

```
  "response":{"numFound":6000,"start":0,"numFoundExact":true,"docs":[
      {
        "id":"igsn:ARF000IW7",
        "isb_core_id":"https://data.isamples.org/digitalsample/igsn/ARF000IW7",
        "indexUpdatedTime":"2021-08-14T00:03:41.240Z",
        "sourceUpdatedTime":"2007-09-20T12:00:00Z",

  "response":{"numFound":27072,"start":0,"numFoundExact":true,"docs":[
      {
        "id":"ark:/28722/k23x8ph6s",
        "isb_core_id":"metadata/28722/k23x8ph6s",
        "indexUpdatedTime":"2021-08-14T00:07:42.870Z",
        "sourceUpdatedTime":"2021-07-02T23:44:20Z",
```

Next step would be to use the field to do a partial import.  Unfortunately we'll need to do another full import of all the records in order to use this moving forward.

File: isb_lib/core.py [hunk 1]
```diff
@@ -129,16 +129,7 @@ def _shouldAddMetadataValueToSolrDoc(metadata: typing.Dict, key: typing.AnyStr)
...
-def coreRecordAsSolrDoc(coreMetadata: typing.Dict) -> typing.Dict:
-    """
-    Args:
-        coreMetadata: the iSamples Core metadata dictionary
-
-    Returns: The coreMetadata in solr document format, suitable for posting to the solr JSON api
-    (https://solr.apache.org/guide/8_1/json-request-api.html)
-    """
-
+def _coreRecordAsSolrDoc(coreMetadata: typing.Dict) -> typing.Dict:
```
Comment (dannymandel) [inline isb_lib/core.py:132]: Had to do some refactoring to allow a unit test to be able to still call this.

File: isb_lib/core.py [hunk 2]
```diff
@@ -176,6 +170,24 @@ def coreRecordAsSolrDoc(coreMetadata: typing.Dict) -> typing.Dict:
...
+    (https://solr.apache.org/guide/8_1/json-request-api.html)
+    """
+    coreMetadata = transformer.transform()
+
+    last_updated = transformer.last_updated_time()
+    if last_updated is not None:
+        date_time = parsed_date(last_updated)
+        if date_time is not None:
+            coreMetadata["sourceUpdatedTime"] = datetimeToSolrStr(date_time)
+    return _coreRecordAsSolrDoc(coreMetadata)
```
Comment (dannymandel) [inline isb_lib/core.py:188]: New wrapper method so we can automatically insert the new column in the SOLR schema.  I didn't add it to the transformed metadata as it wasn't part of the original spec, and 2 of the 4 collections don't have it.

File: scripts/sesar_things.py [hunk 3]
```diff
@@ -174,7 +174,9 @@ def loadRecords(ctx, max_records):
     try:
         oldest_record = None
         res = (
-            session.query(igsn_lib.models.thing.Thing)
+            session.query(igsn_lib.models.thing.Thing).filter(
+            igsn_lib.models.thing.Thing.authority_id == isb_lib.sesar_adapter.SESARItem.AUTHORITY_ID
+            )
```
Comment (dannymandel) [inline scripts/sesar_things.py:179]: This was another bug I had fixed locally when testing something else out and never pushed.

File: solr_schema_init/create_isb_core_schema.py [hunk 4]
```diff
@@ -84,8 +84,17 @@ def addDynamicField(dynamic_field_dict: typing.Dict):
     res = requests.post(f"{SOLR_API}schema", headers=headers, data=data)
     pj(res.json())
 
+#############
+# Internal iSamples bookkeeping columns
 # createField("isb_core_id", "string", True, True, None)
 # createField("source", "string", True, True, None)
+# The time the record was last updated in the source db
```
Comment (dannymandel) [inline solr_schema_init/create_isb_core_schema.py:91]: New fields

File: tests/test_core.py [hunk 5]
```diff
@@ -22,7 +22,7 @@ def test_IdentifierIterator(max_entries, expected_outcome):
 
 def _try_to_add_solr_doc(core_doc_str):
     core_doc = json.loads(core_doc_str)
-    solr_dict = isb_lib.core.coreRecordAsSolrDoc(core_doc)
+    solr_dict = isb_lib.core._coreRecordAsSolrDoc(core_doc)
```
Comment (dannymandel) [inline tests/test_core.py:25]: Update the test to call the new method

File: tests/test_core.py [hunk 6]
```diff
@@ -60,7 +60,7 @@ def test_coreRecordAsSolrDoc():
 }    
     """
     solr_dict = _try_to_add_solr_doc(core_doc_str)
-    assert "producedBy_samplingSite_location_latlon" in solr_dict
+    assert "producedBy_samplingSite_location_ll" in solr_dict
```
Comment (dannymandel) [inline tests/test_core.py:63]: This test was broken by the change to use the RPT fields.  Long-term we need to get some kind of CI setup going on GitHub.

**Review by datadavev — APPROVED** [review]

**Comment by dannymandel** [comment]
This version of things is working on Mars.

**Pull request merged**